.github/scripts/common/header_checker.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+
+import os, sys, re
+from argparse import ArgumentParser
+from difflib import unified_diff
+from json import load
+
+def dprint(msg):
+    print('[DEBUG]: %s' % str(msg))
+
+class HeaderChecker:
+    def __init__(self, header, padding=1000, ignored_files=[], ignored_ext=[], ignored_patterns=[]):
+        self.padding = padding
+        self.header = header
+
+        self.ignorePatternList = ignored_patterns.copy()
```

Comment:
Is this copy necessary? Why not reference the original list?

Reply:
It's a quirk of python, namely optional function args have static refs. If instead I did ` = ignored_patterns`, then every `instance.ignorePatternList` would ref to the same list. The copy creates a new ref that is unique to the object and not shared

Reply:
Maybe instead of an empty list, they could be set to None and checked against, would the copy be needed then? This is more for my understanding, this code doesn't really need to be optimized, but i have not yet looked to see how long this check runs for